File: src/python/handlers/edb.py
from utils import wind_data_to_dict, handle_error
from WindPy import w


def handle_edb(params: dict) -> dict:
    codes = params["codes"]
    begin_time = params["beginTime"]
    end_time = params["endTime"]
    options = params.get("options", "")

    data = w.edb(codes, begin_time, end_time, options)

    if data.ErrorCode != 0:
        raise RuntimeError(handle_error(data.ErrorCode))

    return wind_data_to_dict(data)

File: src/tools/edb.ts
import { z } from "zod";
import { runBridge } from "../bridge/runner.js";

export const edbTool = {
  name: "wind_edb",
  description: "Query Wind Economic Database (EDB) for macro indicators using Wind w.edb().",
  inputSchema: z.object({
    codes: z.string().describe("Comma-separated EDB indicator codes"),
    beginTime: z.string().describe("Start date in YYYY-MM-DD format"),
    endTime: z.string().describe("End date in YYYY-MM-DD format"),
    options: z.string().optional().describe("Wind options string"),
  }),
  handler: async (params: Record<string, unknown>) => {
    const result = await runBridge({ function: "edb", params });
    if (!result.ok) throw new Error(result.error);
    return result.data;
  },
};
